### PR TITLE
Uploading code and a grading script

### PR DESCRIPTION
**Pull request by cindyrzheng**
Still under construction since the APIs for uploading a file is still unimplemented. Need help also with the API request! Not sure what to send in the body! 

Other than that- added context to be passed from the overall class view landing page to the specific assignment view, will need to probably build on that context later but it works as a start! 
then also added modals for uploading student code and slightly changed the modal for uploading a grading script by adding controlled inputs so we can access all the inputs in the form. Also added the scaffold for the API request (it's totally wrong as of now but as long as I can get the body right, and some dummy class IDs and Assignment IDs we're good!).

### CODE REVIEW

**Review by krashanoff — CHANGES_REQUESTED** [review]
Hey Cindy, this is looking good. I left some feedback in the attached thread.

File: frontend/src/AssignmentView/AddAssignmentModal.tsx [hunk 1]
```diff
@@ -1,35 +1,95 @@
...
+    target: { value: string };
+  }) => setGradingScriptFile(value);
+
+  const onInputDueDate = ({
+    target: { value },
+  }: {
+    target: { value: string };
+  }) => setDueDate(value);
+
+  async function handleGradingScript(data: any) {
```
[the remark this answers is not in the document]
Comment (cindyrzheng) [reply frontend/src/AssignmentView/AddAssignmentModal.tsx:47]: amazing thank uuuu

**Comment by cindyrzheng** [comment]
> Hey Cindy, this is looking good. I left some feedback in the attached thread.

tried addressing all the feedback! pls lmk if this is good!

File: frontend/src/AssignmentView/AddAssignmentModal.tsx [hunk 2]
```diff
@@ -1,35 +1,76 @@
...
+  }
+
+  async function submit(e: FormEvent<HTMLFormElement>) {
     e.preventDefault();
-    // send assignment data to database
-    // set up autograding (?)
-    setError((err) => (err ? "" : `File type is not supported`));
-    nav("/professor/class/");
-  };
+    console.log("hey");
```
Comment (krashanoff) [inline frontend/src/AssignmentView/AddAssignmentModal.tsx:57]:
```suggestion
```

File: frontend/src/AssignmentView/AddAssignmentModal.tsx [hunk 3]
```diff
@@ -1,35 +1,76 @@
...
+        method: "POST",
+        mode: "cors",
+        headers: {
+          Authorization: cookies.jwt,
+        },
+        body: data,
+      }
+    )
+      .then((response) => {
+        if (response.status == 401) throw "Unauthorized";
```
Comment (krashanoff) [inline frontend/src/AssignmentView/AddAssignmentModal.tsx:42]:
```suggestion
        if (response.status === 401) throw new Error("Unauthorized");
```

File: frontend/src/AssignmentView/AddAssignmentModal.tsx [hunk 4]
```diff
@@ -1,35 +1,76 @@
...
-    // set up autograding (?)
-    setError((err) => (err ? "" : `File type is not supported`));
-    nav("/professor/class/");
-  };
+    console.log("hey");
+    const data = new FormData(e.currentTarget);
+    handleGradingScript(data);
+    // TODO: The API call to submit a file is still unimplemented. Will fix when implemented
+    //setError((err) => (err ? "" : `File type is not supported`));
+    //nav("/professor/class/");
```
Comment (krashanoff) [inline frontend/src/AssignmentView/AddAssignmentModal.tsx:62]: Remove. API call is implemented. Rebase on main and uncomment this `nav()` call.

File: frontend/src/AssignmentView/AddAssignmentModal.tsx [hunk 5]
```diff
@@ -1,35 +1,76 @@
...
   const nav = useNavigate();
+  const classId = 1;
+  const assignmentId = 1;
 
   const [show, setShow] = useState(false);
+  const [gradingScriptName, setGradingScriptName] = useState("");
+  const [gradingScriptFile, setGradingScriptFile] = useState("");
+  const [dueDate, setDueDate] = useState("");
   const handleClose = () => setShow(false);
   const handleShow = () => setShow(true);
```
Comment (krashanoff) [inline frontend/src/AssignmentView/AddAssignmentModal.tsx:27]: If these functions are only used once they can be inlined as long as it's readable.

File: frontend/src/AssignmentView/UploadSubmissionModal.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        method: "POST",
+        mode: "cors",
+        headers: {
+          Authorization: cookies.jwt,
+        },
+        body: data,
+      }
+    )
+      .then((response) => {
+        if (response.status == 401) throw "Unauthorized";
```
Comment (krashanoff) [inline frontend/src/AssignmentView/UploadSubmissionModal.tsx:42]:
```suggestion
        if (response.status === 401) throw new Error("Unauthorized");
```

File: frontend/src/AssignmentView/UploadSubmissionModal.tsx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+      });
+  }
+
+  function submit(e: FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    const data = new FormData(e.currentTarget);
+    handleSubmission(data);
+    // TODO: The API call to submit a file is still unimplemented. Will fix when implemented
+    setError((err) => (err ? "" : `File type is not supported`));
+    //nav("/professor/class/");
```
Comment (krashanoff) [inline frontend/src/AssignmentView/UploadSubmissionModal.tsx:61]: Again, the call is implemented. If you mean to describe the behavior being unimplemented, please note it here.

**Review by krashanoff — COMMENTED** [review]
Looks good following nitpicks.

File: frontend/src/AssignmentView/AddAssignmentModal.tsx (hunk 2 above)
Comment (cindyrzheng) [reply frontend/src/AssignmentView/AddAssignmentModal.tsx:57]: oops LOL

File: frontend/src/AssignmentView/UploadSubmissionModal.tsx (hunk 7 above)
Comment (cindyrzheng) [reply frontend/src/AssignmentView/UploadSubmissionModal.tsx:61]: haha oops i was rlly tired writing this and missed a lot of small things, my b

File: frontend/src/AssignmentView/AddAssignmentModal.tsx (hunk 5 above)
Comment (cindyrzheng) [reply frontend/src/AssignmentView/AddAssignmentModal.tsx:27]: only handleshow used 1 time only so I changed that!

**Review by krashanoff — APPROVED** [review]

**Pull request merged**